shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/Spark310Shims.scala

```diff
@@ -298,4 +298,15 @@ class Spark310Shims extends Spark301Shims {
     val shuffleOrigin = cpuShuffle.map(_.shuffleOrigin).getOrElse(ENSURE_REQUIREMENTS)
     GpuShuffleExchangeExec(outputPartitioning, child, shuffleOrigin)
   }
+
+  override def sortOrderChildren(s: SortOrder): Seq[Expression] = s.children
+
+  override def sortOrder(
```

Comment:
I think I've missed why the three argument `sortOrder()` method isn't in the `SparkShims` base class.

Reply:
That's the method that each specific shim instance needs to override.  The two-argument version is just syntactic sugar to call the three argument version which must be implemented in the shims.  If the three argument version were to be placed in the SparkShims class then we wouldn't need to shim it at all.  :smile:

Reply:
I was referring to the fact that the code seemed identical across the derived Shims classes. But I see that your reply to my other comment applies here as well.